### PR TITLE
test(inference): add deterministic/stochastic/VI end-to-end smoke

### PR DESCRIPTION
**Pull request by TobiBu**
## Summary
- add synthetic end-to-end smoke coverage for deterministic optimization, stochastic optimization with explicit noise key, and VI cube optimization
- validate numerical finiteness and objective/loss improvement trends in one compact test
- document smoke-test invocation in inference workflow docs

## Validation
- pre-commit passed on changed files
- compileall passed for new test module

### CODE REVIEW

File: tests/test_inference_e2e_smoke.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        pipeline=pipeline,
+        params_init=params_init,
+        static_data=static_data,
+        target=target,
+        learning_rate=0.1,
+        max_steps=20,
+        tol=1e-8,
+        noise_key=jax.random.PRNGKey(7),
+    )
+    assert jnp.isfinite(sto_result.final_loss)
```
[the remark this answers is not in the document]
Comment (TobiBu) [reply tests/test_inference_e2e_smoke.py:95]: @copilot apply changes based on [this feedback](https://github.com/AstroAI-Lab/rubix/pull/214#discussion_r3113582100)
